tools/rapids-upload-to-anaconda

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+# A utility script that find all of the conda packages within a folder and
+# uploads them to Anaconda.org
+# Positional Arguments:
+#   1) a directory path to search for conda packages
+set -e
+
+SEARCH_DIR="$1"
+
+if [ -z "${SEARCH_DIR}" ]; then
+  echo "Please provide a search directory."
+  exit 1
+fi
+
+if [ ! -d "${SEARCH_DIR}" ]; then
+  echo "Directory ${SEARCH_DIR} does not exist."
+  exit 1
+fi
+
+PKGS_TO_UPLOAD=$(find "${SEARCH_DIR}" -name "*.tar.bz2")
+if [ -z "${PKGS_TO_UPLOAD}" ]; then
+  echo "Couldn't find any packages to upload in: ${SEARCH_DIR}."
+  ls -l "${SEARCH_DIR}/"*
+  exit 1
+fi
+
+case "${BUILD_TYPE}" in
+  branch)
+    ;&
+  nightly)
+    ;;
+  *)
+    echo 'Only branch builds and nightly builds are uploaded to Anaconda.org'
+    exit 1
+    ;;
+esac
+
+CONDA_USERNAME="rapidsai-nightly"
+CONDA_TOKEN="${NIGHTLY_CONDA_TOKEN}"
+if [[ "${GIT_BRANCH}" == "main" ]]; then
+  CONDA_USERNAME="rapidsai"
```

Comment:
You should be able to override this. Some of our packages are uploaded to the `nvidia` channel from the main branch.

Reply:
I wrote this script initially. In hindsight, do we need to keep around the `CONDA_USERNAME` variable at all? It's only used in the `-u` flag for the `anaconda upload` command. And from the docs below, it says the following about the `-u` flag usage:

```
-u USER, --user USER  User account or Organization, defaults to the current
                        user
```

- https://docs.anaconda.com/anacondaorg/commandreference/#upload

If the upload command defaults to the current user, why do we need to override it at all with the `-u` flag?

Reply:
Interactions between `-t TOKEN` and `upload -u USER`: https://github.com/Anaconda-Platform/anaconda-client/issues/456

Reply:
We decided that:
* We won't do any logic to set or decide CONDA_TOKEN in the script
* The caller must set `CONDA_TOKEN=${whatever_i_want}`. The vision would be that we could have RAPIDSAI organization secrets such as `NVIDIA_CONDA_TOKEN`, `RAPIDS_NIGHTLY_TOKEN`, etc., each one corresponding to the appropriate channel on Anaconda
* We won't pass any override of `-u ${CONDA_USER}`, and instead use the user from the specified auth token